docarray/array/doc_vec/doc_vec.py

```diff
@@ -116,64 +132,113 @@ def __init__(
-                            )
-                        tf_stack.append(val.tensor)
 
-                    stacked: tf.Tensor = tf.stack(tf_stack)
-                    tensor_columns[field_name] = TensorFlowTensor(stacked)
+                    if first_doc_is_none:
+                        _check_none_field()
+                        tensor_columns[field_name] = None
+                    else:
+                        tf_stack = []
+                        for i, doc in enumerate(docs):
+                            val = getattr(doc, field_name)
+                            _check_field_not_none(field_name, doc)
+                            tf_stack.append(val.tensor)
+
+                        stacked: tf.Tensor = tf.stack(tf_stack)
+                        tensor_columns[field_name] = TensorFlowTensor(stacked)
 
                 elif issubclass(field_type, AbstractTensor):
-                    tensor = getattr(docs[0], field_name)
-                    column_shape = (
-                        (len(docs), *tensor.shape)
-                        if tensor is not None
-                        else (len(docs),)
-                    )
-                    tensor_columns[field_name] = field_type._docarray_from_native(
-                        field_type.get_comp_backend().empty(
-                            column_shape,
-                            dtype=tensor.dtype if hasattr(tensor, 'dtype') else None,
-                            device=tensor.device if hasattr(tensor, 'device') else None,
+                    if first_doc_is_none:
+                        _check_none_field()
+                        tensor_columns[field_name] = None
+                    else:
+                        tensor = getattr(docs[0], field_name)
+                        column_shape = (
+                            (len(docs), *tensor.shape)
+                            if tensor is not None
+                            else (len(docs),)
+                        )
+                        tensor_columns[field_name] = field_type._docarray_from_native(
+                            field_type.get_comp_backend().empty(
+                                column_shape,
+                                dtype=tensor.dtype
+                                if hasattr(tensor, 'dtype')
+                                else None,
+                                device=tensor.device
+                                if hasattr(tensor, 'device')
+                                else None,
+                            )
                         )
-                    )
-
-                    for i, doc in enumerate(docs):
-                        val = getattr(doc, field_name)
-                        if val is None:
-                            val = tensor_type.get_comp_backend().none_value()
 
-                        cast(AbstractTensor, tensor_columns[field_name])[i] = val
+                        for i, doc in enumerate(docs):
```

Comment:
another iteration over al the data :/

Reply:
actually no. this happen only once I think. if the first data point is None I check that this is the case for the rest